lockbox-ios/Common/Extensions/Date+.swift

```diff
@@ -0,0 +1,16 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+
+extension Date {
+    public init?(iso8601DateString: String) {
+        let trimmedIsoString = iso8601DateString.replacingOccurrences(of: "\\.\\d+", with: "", options: .regularExpression)
+        if let date = ISO8601DateFormatter().date(from: trimmedIsoString) {
+            self = date
+        } else {
+            return nil
+        }
```

Comment:
Nit: Could you document some of the sample strings this is expected to handle?

Reply:
Does the test count as documentation here? Or are you saying that I could include more cases in the spec file to be more thorough?